pandas/tslib.pyx

```diff
@@ -707,12 +708,28 @@ NaT = NaTType()
 
 iNaT = util.get_nat()
 
-
```

Comment:
maybe blow away ``_checknull_with_nat``(and replace with ``_check_all_nulls``  and see if any timings are affected?